install_provider.sh

```diff
@@ -0,0 +1,294 @@
+        default upgrade;
+        ""      close;
+    }
+
+    # Load configs
+    include /etc/nginx/conf.d/*.conf;
+
+    # example.com
+    server {
+        listen                               8545${ERIGON_USE_SSL:+ ssl http2};
+        listen                               [::]:8545${ERIGON_USE_SSL:+ ssl http2};
+        server_name                          $ERIGON_HOSTNAME;
+
+        ${ERIGON_USE_SSL:# SSL}
+        ${ERIGON_USE_SSL:+ssl_certificate                      /etc/letsencrypt/live/$ERIGON_HOSTNAME/fullchain.pem;}
+        ${ERIGON_USE_SSL:+ssl_certificate_key                  /etc/letsencrypt/live/$ERIGON_HOSTNAME/privkey.pem;}
+        ${ERIGON_USE_SSL:+ssl_trusted_certificate              /etc/letsencrypt/live/$ERIGON_HOSTNAME/chain.pem;}
+
+        # security headers
+        add_header X-XSS-Protection          "1; mode=block" always;
+        add_header X-Content-Type-Options    "nosniff" always;
+        add_header Referrer-Policy           "no-referrer-when-downgrade" always;
+        add_header Content-Security-Policy   "default-src 'self' http: https: data: blob: 'unsafe-inline'; frame-ancestors 'self';" always;
+        add_header Strict-Transport-Security "max-age=31536000" always;
+
+        auth_basic                           "erigon";
+        auth_basic_user_file                 erigon_htpasswd;
+
+        # reverse proxy
+        location / {
+            proxy_pass                         http://127.0.0.1:8545;
+            proxy_http_version                 1.1;
+            proxy_cache_bypass                 \$http_upgrade;
+
+            # Proxy headers
+            proxy_set_header Upgrade           \$http_upgrade;
+            proxy_set_header Connection        \$connection_upgrade;
+            proxy_set_header Host              \$host;
+            proxy_set_header X-Real-IP         \$remote_addr;
+            proxy_set_header X-Forwarded-For   \$proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto \$scheme;
+            proxy_set_header X-Forwarded-Host  \$host;
+            proxy_set_header X-Forwarded-Port  \$server_port;
+
+            # Proxy timeouts
+            proxy_connect_timeout              60s;
+            proxy_send_timeout                 60s;
+            proxy_read_timeout                 60s;
+        }
+
+        # gzip
+        gzip            on;
+        gzip_vary       on;
+        gzip_proxied    any;
+        gzip_comp_level 6;
+        gzip_types      text/plain text/css text/xml application/json application/javascript application/rss+xml application/atom+xml image/svg+xml;
+    }
+}
+EOF
+
+nginx -t || die "nginx doesn't like it's config"
+systemctl restart nginx
+systemctl enable --now reload-nginx.path
+
+# ========= yagna =========
+
+# install erigon runtime before yagna itself
+mkdir -p "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon"
+cat >"${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon.json" <<EOF
```

Comment:
There are multiple files generated like this and for the services I think it's OK. I think runtime related files we'd be more happy to have them outside of the script included separately. I can imagine that regarding our handbook effort we'd like to point some file and explain something inside it... WDYT @etam ?

I think we can prepare a folder inside a repo containing all runtime files and `curl` them in the script. Later we can simplify link (e.g. point at master or other branch - but it's important right now...